eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Mgmt/src/SystemObjectTypeVisitor.cs

```diff
@@ -0,0 +1,175 @@
+            {
+                Update(baseSystemType, type);
+            }
+            return type;
+        }
+
+        protected override TypeProvider? VisitType(TypeProvider type)
+        {
+            if (type is ModelFactoryProvider modelFactory)
+            {
+                UpdateModelFactory(modelFactory);
+            }
+
+            if (type is SystemObjectTypeProvider systemType)
+            {
+                UpdateNamespace(systemType);
+            }
+
+            if (type is ModelProvider model && model is not SystemObjectTypeProvider && model.BaseModelProvider is SystemObjectTypeProvider baseSystemType)
+            {
+                Update(baseSystemType, model);
+            }
+            return type;
+        }
+
+        // Only this is needed for the current visitor
+        // TODO: The rest of the visitor is temparary workaround for now, once we can generate Azure.ResourceManager from MPG generator, we can remove them
+        private static void UpdateModelFactory(ModelFactoryProvider modelFactory)
+        {
+            var methods = new List<MethodProvider>();
+            foreach (var method in modelFactory.Methods)
+            {
+                var returnType = method.Signature.ReturnType;
+                if (returnType is not null && KnownManagementTypes.IsKnownManagementType($"{returnType.Namespace}.{returnType.Name}"))
+                {
+                    continue;
+                }
+                methods.Add(method);
+            }
+            modelFactory.Update(methods: methods);
+        }
+
+        private static void UpdateNamespace(SystemObjectTypeProvider systemType)
+        {
+            // This is needed because we updated the namespace with NamespaceVisitor in Azure generator earlier
+            systemType.Type.Update(@namespace: systemType._type.Namespace);
+        }
+
+        private static HashSet<ModelProvider> _updated = new();
+        private static void Update(SystemObjectTypeProvider baseSystemType, ModelProvider model)
+        {
+            // Add cache to avoid duplicated update of PreVisitModel and VisitType
+            if (_updated.Contains(model))
+            {
+                return;
+            }
+
+            var rawDataField = CreateRawDataField(model);
+            UpdateSerialization(model);
+
+            UpdateFullConstructor(model, rawDataField);
+
+            var baseSystemPropertyNames = EnumerateBaseModelProperties(baseSystemType);
+            var properties = model.Properties.Where(prop => !baseSystemPropertyNames.Contains(prop.Name));
+
+            model.Update(properties: properties, fields: [.. model.Fields, rawDataField]);
+
+            _updated.Add(model);
+        }
+
+        private static readonly HashSet<string> _methodNamesToUpdate = new(){ "JsonModelCreateCore", "PersistableModelCreateCore", "PersistableModelWriteCore" };
+        private static void UpdateSerialization(ModelProvider model)
```

Comment:
nit: Maybe name the method UpdateSerializationForSystemObject?

Reply:
The whole visitor is for SystemObject, would be redundant to have it in method name.